common/defs.go

```diff
@@ -82,6 +89,76 @@ type PortUpdateMessage struct {
 	Port uint64 `json:"port"`
 }
 
+type Endpoint struct {
+	CidrStr string `json:"cidr,omitempty"`
+	// TODO this can be collapsed into Cidr but needs
+	// work on JSON marshaller/unmarshaller to do that.
+	Cidr      net.IPNet `json:"-"`
+	TenantID  uint64    `json:"tenant_id,omitempty"`
+	TenantExternalID string  `json:"tenant_external_id,omitempty"`
+	SegmentID uint64    `json:"segment_id,omitempty"`
+	SegmentExternalID string `json:"segment_external_id,omitempty"`
+}
+
+const (
+	PolicyDirectionIngress = "ingress"
+	PolicyDirectionEgress  = "egress"
+)
+
+type PortRange [2]uint
+
+type Rule struct {
+	Protocol   string
+	Ports      []uint
+	PortRanges []PortRange
+	IcmpType   uint
+	IcmpCode   uint
+	Stateful   bool
+}
+
+type Rules []Rule
+
+// Metadata attached to entities for various external environments like openstack/kybernetes
+type Tag struct {
+	Key   string
+	Value string
+}
+
+type Policy struct {
+	Direction   string     `json:"direction,omitempty"`
+	Description string     `json:"description,omitempty"`
+	Name        string     `json:"name,omitempty"`
```

Comment:
There are a number of possibilities. For example, currently our iptables rules names are quite long. Something like:

ROMANA-TxSy-......

It's all very readable. You can with great certainty detect Romana related rules, you can see the tenant and segment ID right there, followed by the policy name.

We could certainly shorten 'ROMANA' to 'ROM' or something. We don't technically need the tenant and segment ID in the name, but it's nice to have it there.

So, there are options to reduce it.

We could take ROMANA, change it to ROM and thus add 3 characters. Policy names of 9 or less characters can now be printed as is, while longer ones may be shortened to either a hex ID, or maybe the first 5 characters with a 4 character hex number afterwards. That way you can at least guess what policy it is right from the visible name in the iptables rules.